### PR TITLE
Kyv perfomance metrics 

### PR DESCRIPTION
**Pull request by RingerJK**

### CODE REVIEW

File: MapboxVision/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,3 +1,4 @@
-<manifest package="com.mapbox.vision">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.mapbox.vision">
```
Comment (yunikkk) [inline MapboxVision/src/main/AndroidManifest.xml]: ?

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt [hunk 2]
```diff
@@ -161,6 +163,7 @@ object VisionManager : BaseVisionManager {
         this.application = application
         this.mapboxToken = mapboxToken
         PreferencesManager.appContext = application
+        metricsManager = MetricsManager.Impl(application)
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: Is it really global or we can initialize it in `create`?

Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: it's not global and `MetricManager` doesn't have `destroy` method, but it has `attach`/`detach` methods. So I think we can keep it here

File: MapboxVision/src/main/AndroidManifest.xml (hunk 1 above)
Comment (RingerJK) [reply MapboxVision/src/main/AndroidManifest.xml]: revert

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt (hunk 2 above)
Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: But if we initialize it in `init` it will be effectively global and reused forever once Vision is initialized

**Comment by yunikkk** [comment]
Conflicts

**Comment by RingerJK** [comment]
wait for https://github.com/mapbox/mapbox-vision/pull/1305

**Comment by RingerJK** [comment]
@yunikkk PTAL

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt [hunk 3]
```diff
@@ -152,6 +158,7 @@ object VisionManager : BaseVisionManager {
         this.application = application
         this.mapboxToken = mapboxToken
         PreferencesManager.appContext = application
+        performanceProvider = PerformanceProvider.Impl(application).addTo(attachedModules)
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: Can't recall why we need to init it in `init`, would it be better to do in `onCreate`?

Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: didn't get your point in previous comment. I'm agree

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt [hunk 4]
```diff
@@ -292,6 +302,7 @@ object VisionManager : BaseVisionManager {
 
         locationEngine.detach()
         sensorsManager.detach()
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: Mind adding an issue to reuse `Attachable` for other components here, too? Or doing in in this PR?

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt [hunk 5]
```diff
@@ -76,6 +79,9 @@ object VisionManager : BaseVisionManager {
     private lateinit var locationEngine: LocationEngine
     private lateinit var sessionManager: SessionManager
     private lateinit var videoRecorder: VideoRecorder
+    private lateinit var performanceProvider: PerformanceProvider
+
+    private val attachedModules = ArrayList<Attachable>()
```
Comment (yunikkk) [inline MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: Also seems they are `attachable*`, not `attached*`, i'd suggest to rename

**Comment by yunikkk** [comment]
Approve with minor edits

File: MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt (hunk 4 above)
Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: I'd like to avoid do it in this PR, so I'm gonna create issue for it

Comment (yunikkk) [reply MapboxVision/src/main/java/com/mapbox/vision/VisionManager.kt]: K!
